instrumentation-api/src/main/java/io/opentelemetry/instrumentation/api/tracer/async/package-info.java

```diff
@@ -2,4 +2,4 @@
  * Provides implementations of strategies for tracing methods that return asynchronous and reactive
  * values so that the span can be ended when the asynchronous operation completes.
  */
-package io.opentelemetry.javaagent.instrumentation.otelannotations.async;
+package io.opentelemetry.instrumentation.api.tracer.async;
```

Comment:
maybe `.async` -> `.future` or `.promise`

`.async` feels a bit general

Reply:
I'd probably gravitate to `reactive` but that's probably as general as `async` is.  Of your suggestions I think `future` probably fits better in the Java ecosystem.